pkg/agent/flowexporter/connections/connections.go

```diff
@@ -201,16 +203,29 @@ func (cs *connectionStore) addOrUpdateConn(conn *flowexporter.Connection) {
 		metrics.TotalAntreaConnectionsInConnTrackTable.Inc()
 		klog.V(4).Infof("New Antrea flow added: %v", conn)
 		// Add new antrea connection to connection store
-		cs.connections[connKey] = *conn
+		cs.connections[connKey] = conn
 	}
 }
 
 // GetConnByKey gets the connection in connection map given the connection key.
-func (cs *connectionStore) GetConnByKey(flowTuple flowexporter.ConnectionKey) (*flowexporter.Connection, bool) {
+func (cs *connectionStore) GetConnByKey(connKey flowexporter.ConnectionKey) (*flowexporter.Connection, bool) {
 	cs.mutex.Lock()
 	defer cs.mutex.Unlock()
-	conn, found := cs.connections[flowTuple]
-	return &conn, found
+	conn, found := cs.connections[connKey]
+	return conn, found
+}
+
+// SetExportDone sets DoneExport field of connection to true given the connection key.
+func (cs *connectionStore) SetExportDone(connKey flowexporter.ConnectionKey) error {
+	cs.mutex.Lock()
+	defer cs.mutex.Unlock()
+	if conn, found := cs.connections[connKey]; !found {
+		return fmt.Errorf("connection with key %v does not exist in connection map", connKey)
+	} else {
+		conn.DoneExport = true
+		cs.connections[connKey] = conn
```

Comment:
Missed removing the re-assignment here? It's not a costly operation but unnecessary and may create confusion.

Reply:
Thanks for pointing out. Removed.